src/Microsoft.AspNet.Mvc.Core/Controllers/DefaultControllerActionArgumentBinder.cs

```diff
@@ -64,6 +64,28 @@ public class DefaultControllerActionArgumentBinder : IControllerActionArgumentBi
                         nameof(actionContext));
             }
 
+            if (actionDescriptor.BoundProperties.Count == 0 &&
+                actionDescriptor.Parameters.Count == 0)
+            {
+                return Task.FromResult<IDictionary<string, object>>(
+                    new Dictionary<string, object>(StringComparer.Ordinal));
```

Comment:
@YuvalItzchakov - the dictionary is exposed later though a public contract and needs to be mutable. We could try to avoid it with further shenanigans, but that would be a much more impactful change for a smaller benefit.